hpobench/benchmarks/ml_mmfb/histgb_benchmark.py

```diff
@@ -0,0 +1,136 @@
+            variable=CS.UniformIntegerHyperparameter(
+                'n_estimators', lower=100, upper=1000, default_value=1000, log=False
+            )
+        )
+        fidelity2 = dict(
+            fixed=CS.Constant('subsample', value=1),
+            variable=CS.UniformFloatHyperparameter(
+                'subsample', lower=0.1, upper=1, default_value=1, log=False
+            )
+        )
+        ntrees = fidelity1[ntrees_choice]
+        subsample = fidelity2[subsample_choice]
+        return ntrees, subsample
+
+    def init_model(self, config: Dict, fidelity: Dict = None, rng: Union[int, np.random.RandomState, None] = None):
+        """ Function that returns the model initialized based on the configuration and fidelity
+        """
+        rng = self.rng if rng is None else rng
+        model = HistGradientBoostingClassifier(
+            **config,
+            max_iter=fidelity['n_estimators'],  # a fidelity being used during initialization
+            early_stopping=False,
+            random_state=rng
+        )
+        return model
+
+
+class HistGBSearchSpace0Benchmark(HistGBBenchmark):
+    def get_fidelity_space(self, seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        fidelity_space = CS.ConfigurationSpace(seed=seed)
+        fidelity_space.add_hyperparameters(
+            # black-box setting (full fidelity)
+            HistGBBenchmark._get_fidelity_choices(ntrees_choice='fixed', subsample_choice='fixed')
+        )
+        return fidelity_space
+
+
+class HistGBSearchSpace1Benchmark(HistGBBenchmark):
+    def get_fidelity_space(self, seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        fidelity_space = CS.ConfigurationSpace(seed=seed)
+        fidelity_space.add_hyperparameters(
+            # gray-box setting (multi-fidelity) - ntrees
+            HistGBBenchmark._get_fidelity_choices(ntrees_choice='variable', subsample_choice='fixed')
+        )
+        return fidelity_space
+
+
+class HistGBSearchSpace2Benchmark(HistGBBenchmark):
+    def get_fidelity_space(self, seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        fidelity_space = CS.ConfigurationSpace(seed=seed)
+        fidelity_space.add_hyperparameters(
+            # gray-box setting (multi-fidelity) - subsample
+            HistGBBenchmark._get_fidelity_choices(ntrees_choice='fixed', subsample_choice='variable')
+        )
+        return fidelity_space
+
+
+class HistGBSearchSpace3Benchmark(HistGBBenchmark):
+    def get_fidelity_space(self, seed: Union[int, None] = None) -> CS.ConfigurationSpace:
+        fidelity_space = CS.ConfigurationSpace(seed=seed)
+        fidelity_space.add_hyperparameters(
+            # gray-box setting (multi-multi-fidelity) - ntrees + data subsample
+            HistGBBenchmark._get_fidelity_choices(ntrees_choice='variable', subsample_choice='variable')
+        )
+        return fidelity_space
+
+
+__all__ = [HistGBSearchSpace0Benchmark, HistGBSearchSpace1Benchmark,
```

Comment:
The design to bypass the static function definition is nice. I too have some local commits pending rn but it's a slightly different design where I am overloading the function with a different parameter space. One reason I like your design is what I was actually aiming for since we then have just one definition for the parameter ranges, which is always safer and robust to future changes. I think I can safely then forego my local commits wrt this but I'd like to have a bit more structure and interpretability with the Search Space overloading as follows:

* HistGBBenchmark:  `ntrees_choice='variable`, `subsample_choice='variable'` (multi-multi-fidelity version by default)
* HistGBBenchmarkBB:  `ntrees_choice='fixed`, `subsample_choice='fixed'` (black-box version)
* HistGBBenchmarkMF:  `ntrees_choice='variable`, `subsample_choice='fixed'` (multi-fidelity version)

I think this design can hold us in good stead in the long run too. Having a second Mult-fidelity version with `subsample='variable'` and `ntrees='fixed'` is not quite needed I feel. Especially once we collect data for multi-multi-fidelity, all other cases will be subsumed by it. 

This design is what I'll recommend for the other model spaces too!

Reply:
Sounds good. I will change it.